### ISSUE TITLE
[VL] Umbrella tracker for data mismatch issues

### ISSUE THREAD

**Issue by zhouyuan** [oap-project/gluten#4652]
### Backend

VL (Velox)

### Bug description

There are several data mismatch issues either related with operator or functions. Some of the fixes are landed in Gluten, and some are in Velox repo.
We will use this issue to track the status as these are critical for production envs. 

### Spark version

None

**Comment by FelixYBW** [comment]
https://github.com/oap-project/gluten/issues/4587

Currently we disabled all complex data read

**Comment by zhouyuan** [comment]
https://github.com/oap-project/gluten/pull/4818

**Comment by zhouyuan** [comment]
https://github.com/oap-project/gluten/pull/4872

**Comment by FelixYBW** [comment]
I met an issue today that a simple filter match=1 caused result mismatch. The query has only a parquet scan and a filter. scan column is string, filter column is int.

@rui-mo Have we met this issue before?

**Comment by rui-mo** [comment]
@FelixYBW Will check if I can reproduce.

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/4891

**Comment by rui-mo** [comment]
> I met an issue today that a simple filter match=1 caused result mismatch. The query has only a parquet scan and a filter. scan column is string, filter column is int.
> 
> @rui-mo Have we met this issue before?

@FelixYBW I tried a query like this, but did not reproduce. Maybe more details are needed.

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/4928

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/4930

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/4947

**Comment by FelixYBW** [comment]
3 issues we met:

1.  parquet scan + filter pushdown wrongly return "", should return null. Fixed by https://github.com/facebookincubator/velox/pull/9129
2. distinct hash agg + spill returned duplicated keys.
3. max_by function return wrong result

**Comment by ulysses-you** [comment]
> 2. distinct hash agg + spill returned duplicated keys.

@FelixYBW Has this issue not been fixed by https://github.com/apache/incubator-gluten/pull/4443 ?

**Comment by FelixYBW** [comment]
> @FelixYBW Has this issue not been fixed by #4443 ?

No, it's tested from main branch. A new issue

**Comment by FelixYBW** [comment]
> No, it's tested from main branch. A new issue

https://github.com/facebookincubator/velox/issues/9219

**Comment by FelixYBW** [comment]
> 3. max_by function return wrong result

@yma11 Did you submit a fix to the issue?

**Comment by yma11** [comment]
> > 3. max_by function return wrong result
> 
> @yma11 Did you submit a fix to the issue?

Not yet. Only have pushed to golden branch and will submit one in Velox upstream.

**Comment by NEUpanning** [comment]
#5253

**Comment by FelixYBW** [comment]
> #5253

Looks the issue of get_json_object. @PHILO-HE maybe we need a fully tests of json functions, like the re2.

**Comment by PHILO-HE** [comment]
> > #5253
> 
> Looks the issue of get_json_object. @PHILO-HE maybe we need a fully tests of json functions, like the re2.

@FelixYBW, I will do that. Thanks!

**Comment by FelixYBW** [comment]
https://github.com/apache/incubator-gluten/issues/5248

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/5366

**Comment by FelixYBW** [comment]
> #5366

UPdated desc. thank you. do you know which function (cast, avg, round ) caused the issue?

**Comment by FelixYBW** [comment]
#5372

**Comment by yma11** [comment]
> > > 3. max_by function return wrong result
> > 
> > 
> > @yma11 Did you submit a fix to the issue?
> 
> Not yet. Only have pushed to golden branch and will submit one in Velox upstream.

@FelixYBW This fix should be done at cpp side. The formal fix is in [PR](https://github.com/apache/incubator-gluten/pull/5544). Can you help review it?

**Comment by FelixYBW** [comment]
> @FelixYBW This fix should be done at cpp side. The formal fix is in [PR](https://github.com/apache/incubator-gluten/pull/5544). Can you help review it?

Is it a Gluten issue? I'd think veox has some bug here.

**Comment by yma11** [comment]
> > @FelixYBW This fix should be done at cpp side. The formal fix is in [PR](https://github.com/apache/incubator-gluten/pull/5544). Can you help review it?
> 
> Is it a Gluten issue? I'd think veox has some bug here.

Yes. I think so. It's caused by the additional projects we added before/after shuffle. The logic of partial/final handle in Velox upstream has no problem. The ideal way is to add struct support for shuffle in Gluten so that we can remove the hack.

**Comment by zjuwangg** [comment]
https://github.com/apache/incubator-gluten/issues/5682

**Comment by NEUpanning** [comment]
#5701

**Comment by PHILO-HE** [comment]
> @PHILO-HE Any update of the issues here?

@FelixYBW, Some were actually fixed. Just updated the list. Will fix or seek help to fix other issues.

**Comment by kecookier** [comment]
#6224

**Comment by wang-zhun** [comment]
`VL` cannot parse special characters in `get_json_object`. While #5303 addressed control characters in `CH`, I found that other ASCII characters also cannot be handled in VL. This can be reproduced with the following SQL.

```
with tb as (
select
    concat('{"a": "2", "b": "', string (unhex(id)), '"}') as js,
    string(unhex(id)) as st,
    id as ascii_num
FROM range(0, 255, 1, 10)
)
select 'gluten', ascii_num,st,get_json_object(js,  '$.a') from (select /*+ repartition(2) */ * from tb)
union all 
select 'vanilla', ascii_num,st,get_json_object(js,  '$.a')  from tb
```

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/6630

**Comment by zml1206** [comment]
#6784

**Comment by jiangjiangtian** [comment]
https://github.com/apache/incubator-gluten/issues/6827

**Comment by zml1206** [comment]
#6828

**Comment by FelixYBW** [comment]
> #6845

Closed. Not a bug.

**Comment by jiangjiangtian** [comment]
https://github.com/apache/incubator-gluten/issues/7082

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/7069

**Comment by ccat3z** [comment]
https://github.com/apache/incubator-gluten/issues/7109

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/issues/7194

**Comment by FelixYBW** [comment]
#7494

**Comment by FelixYBW** [comment]
#7730

**Comment by wForget** [comment]
#7749

**Comment by wForget** [comment]
#7777

**Comment by NEUpanning** [comment]
#8184

**Comment by NEUpanning** [comment]
#8227

**Comment by NEUpanning** [comment]
#8754

**Comment by NEUpanning** [comment]
#8787

**Comment by ccat3z** [comment]
https://github.com/apache/incubator-gluten/issues/8959 cast(float|double as string)

**Comment by FelixYBW** [comment]
#9732
